pkg/graphql/fields/generator.go

```diff
@@ -6,12 +6,10 @@ import (
 	"github.com/jensneuse/graphql-go-tools/pkg/operationreport"
 )
 
-type data struct {
-	Types []Type
-}
+type RequestFields map[string]struct{}
```

Comment:
This shouldn't be a map as we discussed, creating a map consumes more time according to array

Reply:
in this case, it should be a map, a request will not contain too many fields, but types could be duplicated

and it simplifies fields comparison

Reply:
search in a map is fast but
create map + search the map > create array + search the array
in terms of complexity in each request.

Reply:
if we want to simplify something, it should be `restricted_fields` of session. We will create it once and will search on it.

Reply:
When we talk about request it should be fine
As numbers here will be for example 10 types per request with 100 fields
so we will not have benefit from using arrays

And with arrays, while collection fields, we will have to do additional checks:
- do we already have such type in array?
- do we already have such field in array?

So actual performance will not differ so much

Reply:
With maps comparison here is straightforward 
https://github.com/TykTechnologies/graphql-go-tools/pull/28/files#diff-e614add267e1e43499221e950455b7faR25

and field collection here is much simpler
https://github.com/TykTechnologies/graphql-go-tools/pull/28/files#diff-61e8b5245eebf03aae8d0886b0e42dd9R48

Reply:
I know map usage is simpler than array. However, don't we normalize a request ? Why may a requeest have field or type duplications?

Reply:
for example 


hero_with_aliases.query from starwars package
```graphql
{
    empireHero: hero {
        name
    }
    jediHero: hero {
        name
    }
}
```

or 
```graphql
person(id: string) {
   name
   friends {
     name
   }
}
```
Return result of Person will be of type Person
But Friends will have return type [Person] which underlying type is also Person
so it will have same restrictions

Normalization will simplify ast document, but it doesn't change what you have requested